code/game/objects/items/devices/data_detector.dm

```diff
@@ -43,22 +42,19 @@
 		var/detected
 		for(var/DT in objects_to_detect)
 			if(istype(I, DT))
-				if(!detect_empty_vial_boxes && istype(I, /obj/item/storage/fancy/vials/random))
-					if(!I.contents)
-						continue
-				detected = TRUE
+				if((istype(I, /obj/item/storage/fancy/vials/random) && !I.contents.len) || (istype(I, /obj/item/reagent_container/glass/beaker/vial/random) && !I.reagents.total_volume))
+					break //We don't need to ping already looted containers
+				else
+					detected = TRUE
```

Comment:
```suggestion
				detected = TRUE
```
The break/continue would already not allow it to hit this code if you were in the if case.